src/DataReader/parse_queries.rs

```diff
@@ -26,19 +26,19 @@ pub fn parse_to_query(query: &str) -> Vec<Query> {
         .collect()
 }
 
-pub fn parse_to_expression_tree(edge_attribute_str: &str) -> Vec<QueryExpression> {
-    let mut pairs = QueryParser::parse(Rule::queries, edge_attribute_str)
-        .unwrap_or_else(|e| panic!("Could not parse as rule with error: {}", e));
+pub fn parse_to_expression_tree(edge_attribute_str: &str) -> Result<Vec<QueryExpression>, String> {
+    let mut pairs = match QueryParser::parse(Rule::queries, edge_attribute_str) {
+        Ok(_) => QueryParser::parse(Rule::queries, edge_attribute_str).unwrap(),
```

Comment:
Seems to be calling parse twice. Should only call it once

Reply:
@henneboy fix it

Reply:
Okay puvi fixed it